src/mlpack/core/data/load_utilities.hpp

```diff
@@ -0,0 +1,490 @@
+/**
+ * @file core/data/load_impl.hpp
+ * @author Ryan Curtin
+ * @author Omar Shrit
+ * @author Gopi Tatiraju
+ *
+ * Implementation of templatized load() function defined in load.hpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_LOAD_UTILITIES_HPP
+#define MLPACK_CORE_DATA_LOAD_UTILITIES_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <string>
+
+#include "dataset_mapper.hpp"
+#include "detect_file_type.hpp"
+#include "format.hpp"
+
+namespace mlpack {
+namespace data {
+
+namespace details{
+
+template<typename Tokenizer>
+std::vector<std::string> ToTokens(Tokenizer& lineTok)
+{
+  std::vector<std::string> tokens;
+  std::transform(std::begin(lineTok), std::end(lineTok),
+                 std::back_inserter(tokens),
+                 [&tokens](std::string const &str)
+  {
+    std::string trimmedToken(str);
+    Trim(trimmedToken);
+    return std::move(trimmedToken);
+  });
+
+  return tokens;
+}
+
+inline
+void TransposeTokens(std::vector<std::vector<std::string>> const &input,
+                     std::vector<std::string>& output,
+                     size_t index)
+{
+  output.clear();
+  for (size_t i = 0; i != input.size(); ++i)
+  {
+    output.emplace_back(input[i][index]);
+  }
+}
+} // namespace details
+
+/**
+ * All possible DataOptions grouped under one class.
+ * This will allow us to have consistent data API for mlpack. If new data
+ * options might be necessary, then they should be added in the following.
+ */
+class DataOptions
```

Comment:
Might be better to put this in a file of its own for people searching through the codebase: `data_options.hpp`.